### PR TITLE
Update github.com/quic-go/quic-go dependency

### LINKED ISSUE #2783: Latest (v0.33.2) `go-libp2p` not compatible with latest (v0.8.0) `quic-go/webtransport-go`
Having the following in `go.mod`

```
	github.com/libp2p/go-libp2p v0.33.2
        ...
	github.com/quic-go/quic-go v0.43.0 // indirect
	github.com/quic-go/webtransport-go v0.8.0 // indirect
```

results in

```
../../go/pkg/mod/github.com/libp2p/go-libp2p@v0.33.2/p2p/transport/webtransport/transport.go:206:3: unknown field RoundTripper in struct literal of type webtransport.Dialer
../../go/pkg/mod/github.com/libp2p/go-libp2p@v0.33.2/p2p/transport/webtransport/transport.go:210:4: unknown field QuicConfig in struct literal of type http3.RoundTripper
```

### PR DESCRIPTION
**Pull request by fasmat**
This updates the `quic-go` dependency to the newest version. The `webtransport.Dialer` interface changed in a non-backwards compatible way so I had to make some minor changes to the code to be able to use the updated dependency.

Fixes #2783

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,8 +11,8 @@ require (
 	github.com/flynn/noise v1.1.0
 	github.com/google/gopacket v1.1.19
 	github.com/gorilla/websocket v1.5.0
-	github.com/hashicorp/golang-lru/arc/v2 v2.0.5
-	github.com/hashicorp/golang-lru/v2 v2.0.5
+	github.com/hashicorp/golang-lru/arc/v2 v2.0.7
+	github.com/hashicorp/golang-lru/v2 v2.0.7
```
Comment (marten-seemann) [inline go.mod:15]: Many of these changes to go.mod seem unrelated, and should not be folded into this PR.

Comment (fasmat) [reply go.mod]: I reverted dependency upgrades that were not directly related to `quic-go`

File: go.mod [hunk 2]
```diff
@@ -112,7 +112,7 @@ require (
 	github.com/pion/turn/v2 v2.1.4 // indirect
 	github.com/pkg/errors v0.9.1 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
-	github.com/prometheus/common v0.47.0 // indirect
```
Comment (marten-seemann) [inline go.mod:115]: Same here?

Comment (fasmat) [reply go.mod]: Updated

**Review by marten-seemann — APPROVED** [review]
lgtm, but I'll let the libp2p team take it from here

**Comment by fasmat** [comment]
After seeing most tests fail I updated the dependencies of the modules in the `examples` and `test-plan` folders. I missed them in my initial submission.

**Comment by MarcoPolo** [comment]
We don't use do `replace github.com/libp2p/go-libp2p => ../` in the examples on purpose. It leads to noisier changes and the examples make sense to reference a released version of go-libp2p.

**Comment by fasmat** [comment]
> Why are the example tests failing?

The tests were not failing just `test-plans` because it used a replacement while the examples didn't. I should have only called `go mod tidy` on `test-plans` instead of updating all `go.mod` files in the examples package...
